### ISSUE TITLE
Occasional NPE in AbstractReadContext#close()

### ISSUE THREAD

**Issue by olavloite** [googleapis/java-spanner-jdbc#1453]
In version 2.9.11

"Caused by: java.lang.NullPointerException",
"\com.google.cloud.spanner.AbstractReadContext.close(AbstractReadContext.java:709)",
"com.google.cloud.spanner.SessionPool$AutoClosingReadContext.close(SessionPool.java:541)",
"com.google.cloud.spanner.SessionPool$AutoClosingReadContext$1.close(SessionPool.java:328)",
"com.google.cloud.spanner.SessionPool$AutoClosingReadContext$1.internalNext(SessionPool.java:306)",
"com.google.cloud.spanner.SessionPool$AutoClosingReadContext$1.next(SessionPool.java:277)",

**Comment by Mironiken** [comment]
Hi! We got a lot of such errors on our production. Do you have any ETA?

**Comment by olavloite** [comment]
> Hi! We got a lot of such errors on our production. Do you have any ETA?

Hi @Mironiken, thanks for reaching out. I've actually not been able to reproduce it locally. Do you have a reproduction scenario?

**Comment by Mironiken** [comment]
Nope. I cannot reproduce it locally either. I can create a build with additional logging and run it on the internal environment. I will notify you if there are any results.
Feel free to share any additional thoughts you may have.
